lib/units/month.js

```diff
@@ -0,0 +1,127 @@
+import { createUTC } from "../create/utc";
+
+export function daysInMonth(year, month) {
+    return new Date(Date.UTC(year, month + 1, 0)).getUTCDate();
+}
+
+// FORMATTING
+
+addFormatToken("M", ["MM", 2], "Mo", function () {
+    return this.month() + 1;
+});
+
+addFormatToken("MMM", 0, 0, function (format) {
+    return this.localeData().monthsShort(this, format);
+});
+
+addFormatToken("MMMM", 0, 0, function (format) {
+    return this.localeData().months(this, format);
+});
+
+// ALIASES
+
+addUnitAlias("month", "M");
+
+// PARSING
+
+addRegexToken("M",    match1to2);
+addRegexToken("MM",   match1to2, match2);
+addRegexToken("MMM",  matchWord);
+addRegexToken("MMMM", matchWord);
+
+addParseToken(["M", "MM"], function (input, array) {
+    array[1] = toInt(input) - 1; // TODO: use a constant for MONTH
+});
+
+addParseToken(["MMM", "MMMM"], function (input, array, config, token) {
+    var month = config._locale.monthsParse(input, token, config._strict);
+    // if we didn't find a month name, mark the date as invalid.
+    if (month != null) {
+        array[1] = month; // TODO: use a constant for MONTH
+    } else {
+        config._pf.invalidMonth = input;
+    }
+});
+
+// LOCALES
+
+export var defaultLocaleMonths = 'January_February_March_April_May_June_July_August_September_October_November_December'.split('_');
+export function localeMonths (m) {
+    return this._months[m.month()];
+}
+
+export var defaultLocaleMonthsShort = 'Jan_Feb_Mar_Apr_May_Jun_Jul_Aug_Sep_Oct_Nov_Dec'.split('_');
+export function localeMonthsShort (m) {
+    return this._monthsShort[m.month()];
+}
+
+export function localeMonthsParse (monthName, format, strict) {
+    var i, mom, regex;
+
+    if (!this._monthsParse) {
+        this._monthsParse = [];
+        this._longMonthsParse = [];
+        this._shortMonthsParse = [];
+    }
+
+    for (i = 0; i < 12; i++) {
+        // make the regex if we don't have it already
+        mom = createUTC([2000, i]);
+        if (strict && !this._longMonthsParse[i]) {
+            this._longMonthsParse[i] = new RegExp('^' + this.months(mom, '').replace('.', '') + '$', 'i');
+            this._shortMonthsParse[i] = new RegExp('^' + this.monthsShort(mom, '').replace('.', '') + '$', 'i');
+        }
+        if (!strict && !this._monthsParse[i]) {
+            regex = '^' + this.months(mom, '') + '|^' + this.monthsShort(mom, '');
+            this._monthsParse[i] = new RegExp(regex.replace('.', ''), 'i');
+        }
+        // test the regex
+        if (strict && format === 'MMMM' && this._longMonthsParse[i].test(monthName)) {
+            return i;
+        } else if (strict && format === 'MMM' && this._shortMonthsParse[i].test(monthName)) {
+            return i;
+        } else if (!strict && this._monthsParse[i].test(monthName)) {
+            return i;
+        }
+    }
+}
+
+// MOMENTS
+
+export function setMonth (mom, value) {
+    var dayOfMonth;
+
+    // TODO: Move this out of here!
```

Comment:
This is from [this commit](https://github.com/moment/moment/commit/44f133a62fb92aa1015d65b116cf13abad64a40f#diff-8c9255ccee4354533dcec358fb36d701R2171) :)